### PR TITLE
Removed ServerCommon code from repo + consolidated package dependencies

### PR DESCRIPTION
**Pull request by xavierdecoster**
This PR does a few things:
* Remove ServerCommon code from this repo (NuGet.Services.KeyVault + tests have moved to NuGet/ServerCommon)
* Added NuGet.Services.KeyVault and NuGet.Services.Logging package dependencies
* Upgraded net451 framework dependency to net452 (required for some ServerCommon package dependencies)
* Consolidated package dependencies in the solution

Once the pending changes on NuGet.Services.KeyVault made it into dev, the ServerCommon packages can RTM to v1.0.0 and a simple upgrade on this solution will do (already unlisted the older packages on the nuget-build feed).

cc @scottbommarito @skofman1 @maartenba @ryuyu

### CODE REVIEW

File: src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj [hunk 1]
```diff
@@ -12,6 +12,8 @@
     <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <DefineConstants>TRACE</DefineConstants>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (skofman1) [inline src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj]: can be removed?

**Comment by skofman1** [comment]
:shipit:

**Comment by ryuyu** [comment]
LGTM

**Comment by shishirx34** [comment]
:shipit:

**Comment by scottbommarito** [comment]
LGTM

feel free to merge this before my changes, I will fix my branch when the time to merge comes

File: src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj (hunk 1 above)
Comment (xavierdecoster) [inline src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj]: Side-effect of running Update-Package -Reinstall... no idea why this gets added every single time. Removing :)

**Pull request merged**